consent-ws/src/main/java/org/broadinstitute/consent/http/models/darsummary/DARModalDetailsDTO.java

```diff
@@ -261,7 +261,10 @@ public void setIsTherePurposeStatements(boolean isTherePurposeStatements) {
 
     public void setDatasetDetail(ArrayList<Document> datasetDetail) {
         Map<String, String> datasetDetailMap = new HashMap<>();
-        datasetDetail.forEach((doc) -> datasetDetailMap.put(doc.getString(DarConstants.DATASET_ID),doc.getString("name")));
+        datasetDetail.forEach((doc) -> {
+            String objectId = doc.getString(DarConstants.OBJECT_ID) != null ? doc.getString(DarConstants.OBJECT_ID) : "";
+            datasetDetailMap.put(objectId ,doc.getString("name"));
```

Comment:
Potential for having an empty string as the key in `datasetDetailMap` which doesn't seem like a great idea here. Maybe you want to filter out the docs that don't have a `DarConstants.OBJECT_ID` property?
Also, spacing nit:
```suggestion
            datasetDetailMap.put(objectId, doc.getString("name"));
```